grapple/actions.py

```diff
@@ -391,15 +398,26 @@ class Meta:
 
     # Add any custom fields to node if they are defined.
     if hasattr(cls, "graphql_fields"):
-        for field in cls.graphql_fields:
-            if callable(field):
-                field = field()
+        for item in cls.graphql_fields:
+            has_custom_source: bool = hasattr(item, "field_source")
+
+            if all(
```

Comment:
praise: good sneaking in `all()` :)

I do wonder whether `if not has_custom_source and callable(item):` is more readable though